### PR TITLE
feat: Added support for external plugin directory

### PR DESCRIPTION
**Pull request by MichielDeMey**
**Type:** feature

This PR allow you to specify an external plugins directory, useful for Docker and Kubernetes deployments.

*  There is a related issue?
Resolves https://github.com/verdaccio/verdaccio/issues/532
Based on https://github.com/verdaccio/verdaccio/pull/232 by @idangozlan 

*  Unit or Functional tests are included in the PR
Tested using https://github.com/verdaccio/verdaccio-google-cloud. No additional unit test added.

**Note:** When dealing with plugins with native dependencies, plugins will have to be built on the same base image as the Verdaccio `Dockerfile`.

When using with Kubernetes, an init container can be used to install plugins before startup.
[https://kubernetes.io/docs/concepts/workloads/pods/init-containers/](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)

**Depends on:** https://github.com/verdaccio/flow-types/pull/15

### CODE REVIEW

**Comment by MichielDeMey** [comment]
Note that this PR will probably fail until https://github.com/verdaccio/flow-types/pull/15 is merged and published to the NPM registry.

**Comment by juanpicado** [comment]
@MichielDeMey ready on `@verdaccio/types@3.4.2`

**Comment by MichielDeMey** [comment]
Updated, looks like there are some flow errors related to storage?
```
Cannot assign `self.localStorage.search(...)` to `lstream` because `verdaccio$IUploadTarball` [1] is incompatible with
`verdaccio$IReadTarball`
```

**Comment by juanpicado** [comment]
Ok, I see, we need to merge https://github.com/verdaccio/verdaccio/pull/815 first. cc @ayusharma

**Comment by ayusharma** [comment]
@juanpicado I am on #815.

**Comment by juanpicado** [comment]
@MichielDeMey rebase and have fun 👍

**Comment by MichielDeMey** [comment]
Rebased and all tests are green across the board. ✅

File: src/lib/plugin-loader.js [hunk 1]
```diff
@@ -56,6 +56,21 @@ export default function loadPlugin<T>(
     // try local plugins first
     plugin = tryLoad(Path.resolve(__dirname + '/../plugins', pluginId));
 
+    // custom plugin directory
```
Comment (juanpicado) [inline src/lib/plugin-loader.js]: Please update the comment above, just it's a bit outdated according to the new logic.

**Comment by MichielDeMey** [comment]
Updated `loadPlugin` documentation to reflect new loading order. 👍

**Comment by MichielDeMey** [comment]
Commits have been squashed and message adheres to the commit message rules 🥇

**Comment by MichielDeMey** [comment]
Thanks guys!
I'll try and add some Kubernetes documentation once I've deployed this to our cluster. ⛵️🐳

**Comment by juanpicado** [comment]
@MichielDeMey you should be able to test your new feature with the `beta` tag. It's out already.
https://hub.docker.com/r/verdaccio/verdaccio/tags/

**Comment by MichielDeMey** [comment]
Way ahead of you, already tested and verified working.
Next step is modifying the Helm chart to allow for an init container that installs specified plugins. 👍

**Comment by juanpicado** [comment]
It works for me
https://github.com/verdaccio/docker-examples/tree/master/docker-plugin-external

**Pull request merged**